### PR TITLE
remove capture_autograd_function flag

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136959

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_dynamo/config.py [hunk 1]
```diff
@@ -425,8 +425,6 @@ def default_debug_dir_root():
 # correctness of custom ops.
 only_allow_pt2_compliant_ops = False
 
-capture_autograd_function = True
```
Comment (anijain2305) [inline torch/_dynamo/config.py:428]: You should not delete the flag. The idea is to keep the flag here, but make it a no-op.

**Review by anijain2305 — DISMISSED** [review]
Lets not delete the flag from the config.py. Everything else LGTM.

File: torch/_dynamo/variables/misc.py [hunk 2]
```diff
@@ -658,11 +660,7 @@ def visit(node):
 
         VariableTracker.visit(visit, (args, kwargs))
 
-        if (
-            requires_grad
-            and torch.is_grad_enabled()
-            and config.capture_autograd_function
-        ):
+        if requires_grad and torch.is_grad_enabled():
```
Comment (amjames) [inline torch/_dynamo/variables/misc.py:663]: The flag is no longer participating in the logic here, but it seems like it has not actually been removed. Is it still used somewhere else? 

If not, and here is some downstream issue blocking actual removal, we should at least produce a warning on set if the config option is going to have no effect.

**Review by jansel — CHANGES_REQUESTED** [review]
(edit) nevermind saw discussion above

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**